### PR TITLE
Bugfix: No Author or Location Errors

### PR DESCRIPTION
**Pull request by shabegom**
This PR fixes two issues, one which breaks the plugin and the other is just aesthetic:

1. No Author Error

If there is no author returned by the Readwise api, `author.split` will return a TypeError. This causes the plugin to stop and highlights aren't written to the notes. To fix this, I'm checking if `author` exists, if it doesn't `authors` returns an empty array. Then when writing `authors` I'm checking that the `authors` array has a value.

2. No Location causes "null" links

If the Readwise API doesn't return a location for a highlight the location link shows up as `[null]()` I've added a check for `location` before rendering the location link.

### CODE REVIEW

**Comment by shabegom** [comment]
so my eslint config changed the formatting of the file. LMK if that's going to be an issue and I can reopen the PR. The only code changes are marked with comments.

File: main.ts [hunk 1]
```diff
@@ -25,71 +25,95 @@ export default class ReadwiseMirror extends Plugin {
...
+        last_highlight_at,
+        source_url,
+      } = book;
+      const fileName = `${title.replace(/[<>:"\/\\|?*]+/g, "")}.md`;
+
+      const formattedHighlights = highlights
+        .map((highlight: Highlight) => this.formatHighlight(highlight, book))
+        .join("");
+
+      const authors = author ? author.split(/and |,/) : [];
```
Comment (shabegom) [inline main.ts:83]: checking for author and returning empty array if null

File: main.ts [hunk 2]
```diff
@@ -25,71 +25,95 @@ export default class ReadwiseMirror extends Plugin {
...
 
       let authorStr =
-        authors.length > 1
+        authors[0] && authors?.length > 1
```
Comment (shabegom) [inline main.ts:86]: checking that authors array has at least one element.

File: main.ts [hunk 3]
```diff
@@ -25,71 +25,95 @@ export default class ReadwiseMirror extends Plugin {
...
           ? authors
-              .filter((authorName: string) => authorName.trim() != '')
+              .filter((authorName: string) => authorName.trim() != "")
               .map((authorName: string) => `[[${authorName.trim()}]]`)
-              .join(', ')
-          : `[[${author}]]`;
+              .join(", ")
+          : author
```
Comment (shabegom) [inline main.ts:91]: if no author exists, don't output anything

**Comment by jsonMartin** [comment]
Hey @shabegom , great catch on this issue! I didn't have any case where there was a missing author, so didn't run into this edge case on my end 👍

Didn't have a chance to look at the code yet, but I'll review soon when I get a chance

**Comment by jsonMartin** [comment]
Hi @shabegom , just had a chance to review the code. The fixes look good to me! 👍

As far as the formatting, I'll go ahead and update that on my end in my next update for version 1.0.1. I actually need to do a rewrite to the git history to remove the file `readwiseApi.js` *(.js, not .ts)* file from the repo, so it should be easier to merge your code changes first before proceeding. 

After I update the git history to remove the file, your fork will be incompatible so please remove your old fork and re-fork the new version if you'd like to edit the codebase further.

Thanks for the contribution!

**Pull request merged**